### PR TITLE
Epmcuxda 6879 filters

### PR DESCRIPTION
**Pull request by yanaSelin**
### What does this PR do?

`Add ability of end user to filter observations with single selection dropdown`

### Jira ticket related to this PR

- [EPMCUXDA-6879](https://jira.epam.com/jira/browse/EPMCUXDA-6879)

### Link to branch and/or PR's related

- https://github.com/abahachuk/ptushki-backend/pull/40

#### Type of change

- [x] Adds new features.
- [ ] Changes existing functionality.
- [ ] Deprecates features.
- [ ] Removes features.
- [ ] Fixes any bug.
- [ ] Other: security, build process, infrastructure, tests, docs, etc.

### CODE REVIEW

**Review by maksim-v — APPROVED** [review]

File: src/app/features/observations/models.ts [hunk 1]
```diff
@@ -21,6 +22,20 @@ export interface ObservationData {
   remarks: string;
 }
 
+export interface ObservationFilters {
```
Comment (PaulGiletich) [inline src/app/features/observations/models.ts]: in order to be less verbose, you can create
```

export type DataGridFiltersObj<TData> = {
  [key in keyof TData]?: DataGridFilter[]
};
```

and extend this interface from DataGridFiltersObj<ObservationData>, leaving only overridden fields:
```


export interface ObservationFilters
  extends DataGridFiltersObj<ObservationData> {
  verificationStatus?: (DataGridFilter & { value: boolean })[];
  finder?: (DataGridFilter & { value: UserInfo })[];
}
```

File: src/components/autosuggest/Autosuggest.tsx [hunk 2]
```diff
@@ -52,30 +52,36 @@ export const Autosuggest: FC<IAutosuggest> = ({
...
-        onClick={onSelect}
-        value={item.value}
-        key={item.id}
-        className={`${blockName}__item`}
-      >
-        {item.label}
-      </DropdownItem>
-    ),
+    (item: Item) =>
+      // @ts-ignore
```
Comment (PaulGiletich) [inline src/components/autosuggest/Autosuggest.tsx]: i removed ts-ignore locally and it is ok. 
does it throw anything for you?

File: src/components/table/customisations/TableHeaderRowContent.tsx [hunk 3]
```diff
@@ -1,15 +1,62 @@
...
+  </TableHeaderRow.Content>
+);
+
+export const TableHeaderRowContentConnected = connect(
+  (state: DataGridState, ownProps: { column: Column }) => {
+    const columnFilter = state.filtering.find(
+      f => f.columnName === ownProps.column.name
+    );
+
+    // @ts-ignore
```
Comment (PaulGiletich) [inline src/components/table/customisations/TableHeaderRowContent.tsx]: for better type safety, 

```


    const filters = (state.filters as DataGridFiltersObj<any>)[
      ownProps.column.name
    ];
```

File: src/components/table/DataGrid.tsx [hunk 4]
```diff
@@ -40,14 +40,23 @@ import {
...
 import { TableComponent } from "./customisations/TableComponent";
-import { TableHeaderRowContent } from "./customisations/TableHeaderRowContent";
+import { TableHeaderRowContentConnected } from "./customisations/TableHeaderRowContent";
 import { ToolbarConnected } from "./customisations/ToolbarComponent";
 import { DataGridState } from "./DataGridModels";
 
+export interface DataGridFilter {
+  count: number;
+  value: string | boolean;
+}
```
Comment (PaulGiletich) [inline src/components/table/DataGrid.tsx]: this better be in DataGridModels

File: src/store/effects/observationListEpics.ts [hunk 5]
```diff
@@ -46,7 +56,30 @@ export const requestObservationEpic: Epic<any, any, RootState> = (
...
+) =>
+  action$.pipe(
+    filter(isActionOf([observationsFiltersRequest])),
+    withLatestFrom(state$),
+    switchMap(([, state]) => {
+      return from(
+        ajaxService.makeCall<ObservationFilters>(OBSERVATIONS_FILTERS_ENDPOINT)
+      ).pipe(
+        map(d => observationGridActions.addFilters(d)),
+        // eslint-disable-next-line consistent-return
```
Comment (PaulGiletich) [inline src/store/effects/observationListEpics.ts]: you can just return EMPTY imported from rxjs instead

File: src/components/table/DataGridModels.ts [hunk 6]
```diff
@@ -31,4 +31,5 @@ export interface DataGridState {
   filtering: FilteringRule[];
   sorting: Sorting[];
   search: string;
+  filters: TFilters;
```
Comment (PaulGiletich) [inline src/components/table/DataGridModels.ts]: maybe rename this to availableFilters?
having `filtering` and `filters` props may be confusing when reading code

**Review by PaulGiletich — CHANGES_REQUESTED** [review]
see comments

**Pull request merged**